src/main/java/com/gepardec/mega/service/impl/MonthlyReportServiceImpl.java

```diff
@@ -85,52 +63,121 @@ public class MonthlyReportServiceImpl implements MonthlyReportService {
-        final LocalDate date = Optional.ofNullable(zepService.getEmployee(userId))
-                .flatMap(emp -> Optional.ofNullable(emp.getReleaseDate()))
-                .filter(this::checkReleaseDate)
-                .map(releaseDate -> LocalDate.parse(Objects.requireNonNull(releaseDate)).plusMonths(1))
-                .orElse(null);
-
-        return getMonthendReportForUser(userId, date);
+    public MonthlyReport getMonthEndReportForUser() {
+        Employee employee = employeeService.getEmployee(userContext.getUser().getUserId());
+
+        LocalDate initialDate = getCorrectInitialDateForEndReport(employee);
+
+        MonthlyReport monthlyReport = getMonthEndReportForUser(initialDate.getYear(), initialDate.getMonthValue(), employee);
+        monthlyReport.setInitialDate(initialDate);
+
+        return monthlyReport;
     }
 
-    private boolean checkReleaseDate(String releaseDate) {
-        try {
-            LocalDate.parse(Objects.requireNonNull(releaseDate));
-            return true;
-        } catch (DateTimeParseException ex) {
-            return false;
+    private LocalDate getCorrectInitialDateForEndReport(Employee employee) {
+        LocalDate midOfMonth = LocalDate.now().withDayOfMonth(14);
+        LocalDate now = LocalDate.now();
+        LocalDate firstOfPreviousMonth = now.withMonth(now.getMonth().minus(1).getValue()).withDayOfMonth(1);
+
+        if (now.isAfter(midOfMonth) && isMonthCompletedForEmployee(employee, firstOfPreviousMonth)) {
+            return now;
+        } else {
+            return firstOfPreviousMonth;
         }
     }
 
     @Override
-    public MonthlyReport getMonthendReportForUser(String userId, LocalDate date) {
-        Employee employee = zepService.getEmployee(userId);
+    public MonthlyReport getMonthEndReportForUser(Integer year, Integer month, Employee employee) {
+        LocalDate date = LocalDate.of(year, month, 1);
 
-        // Wenn kein date null, dann 1. Tag von aktuellen Monat (Logik wie bei StepEntryServiceImpl.parseReleaseDate)
-        currentMonthYear = Optional.ofNullable(date).orElse(DateUtils.getFirstDayOfCurrentMonth());
+        if (employee == null) {
+            employee = employeeService.getEmployee(userContext.getUser().getUserId());
+        }
 
-        return buildMonthlyReport(employee,
+        MonthlyReport monthlyReport = buildMonthlyReport(
+                employee,
+                date,
                 zepService.getProjectTimes(employee, date),
                 zepService.getBillableForEmployee(employee, date),
                 zepService.getAbsenceForEmployee(employee, date),
                 stepEntryService.findEmployeeCheckState(employee, date),
-                stepEntryService.findEmployeeInternalCheckState(employee, date));
+                stepEntryService.findEmployeeInternalCheckState(employee, date)
+        );
+
+        if (monthlyReport == null) {
+            monthlyReport = MonthlyReport.builder()
```

Comment:
Method buildMonthlyReport never returns null, so the null check can be removed and the result can be returned immediately